library/src/commonMain/kotlin/com/lagradost/cloudstream3/extractors/Vidmoly.kt

```diff
@@ -26,15 +27,25 @@ open class Vidmoly : ExtractorApi() {
         callback: (ExtractorLink) -> Unit
     ) {
         val headers  = mapOf(
-            "User-Agent"     to "Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Mobile Safari/537.36",
+            "user-agent"     to USER_AGENT,
             "Sec-Fetch-Dest" to "iframe"
         )
-        val script = app.get(
-            url,
-            headers = headers,
-            referer = referer,
-        ).document.select("script")
-            .find { it.data().contains("sources:") }?.data()
+        val newUrl = if(url.contains("/w/"))
+            url.replaceFirst("/w/", "/embed-")+"-920x360.html"
+            else url
+        var script: String? = null;
+        var attemps = 0
+        while (attemps < 10 && script.isNullOrEmpty()){
+            attemps++
+            script = app.get(
+                newUrl,
+                headers = headers,
+                referer = referer,
+            ).document.select("script")
+                .firstOrNull { it.data().contains("sources:") }?.data()
+            if(script.isNullOrEmpty())
+                Thread.sleep(500)
```

Comment:
Thread.sleep? Use delay to not stop execution of concurrent requests. https://stackoverflow.com/questions/61345712/what-is-the-difference-between-delay-and-thread-sleep-in-kotlin

Reply:
Thanks for the review, already replaced by sleep